fhir2dataset/api_caller.py

```diff
@@ -71,21 +71,31 @@ def get_response(self, url: str):
         response = self._get_bundle_response(url)
         self.status_code = response.status_code
         try:
-            self.results = response.json()["entry"]
-        except KeyError as e:
-            logger.info(f"Got a KeyError - There's no {e} key in the json data we received.")
-            logger.debug(f"status code of KeyError response:\n{response.status_code}")
-            logger.debug(f"content of the KeyError response:\n{response.content}")
-        try:
-            self.next_url = None
-            for relation in response.json()["link"]:
-                if relation["relation"] == "next":
-                    self.next_url = relation["url"]
-                    break
+            total = response.json()["total"]
         except KeyError as e:
             logger.info(f"Got a KeyError - There's no {e} key in the json data we received.")
             logger.debug(f"status code of KeyError response:\n{response.status_code}")
             logger.debug(f"content of the KeyError response:\n{response.content}")
+        if total == 0:
+            # no resources match the request
+            self.results = {}
```

Comment:
Self.request is set to None by default. I think that we can simplify the code and do only (be careful to shift the rest of the code in the new if condition):
```
if total != 0:
            try:
                self.results = response.json()["entry"]
```